tsl/src/chunk.c

```diff
@@ -334,6 +336,62 @@ chunk_create_replica_table(PG_FUNCTION_ARGS)
 	PG_RETURN_VOID();
 }
 
+static void
+chunk_api_call_chunk_drop_replica(Chunk *chunk, const char *node_name, Oid serverid)
+{
+	const char *drop_cmd;
+	List *data_nodes;
+
+	/*
+	 * Drop chunk on the data node using a regular "DROP TABLE".
+	 * Note that CASCADE is not required as it takes care of dropping compressed
+	 * chunk (if any).
+	 *
+	 * If there are any other non-TimescaleDB objects attached to this table due
+	 * to some manual user activity then they should be dropped by the user
+	 * before invoking this function.
+	 */
+
+	drop_cmd = psprintf("DROP TABLE %s.%s",
+						quote_identifier(chunk->fd.schema_name.data),
+						quote_identifier(chunk->fd.table_name.data));
+	data_nodes = list_make1((char *) node_name);
+	ts_dist_cmd_run_on_data_nodes(drop_cmd, data_nodes, true);
+
+	/*
+	 * This chunk might have this data node as primary, change that association
+	 * if so. Then delete the chunk_id and node_name association.
+	 */
+	chunk_update_foreign_server_if_needed(chunk->fd.id, serverid);
+	ts_chunk_data_node_delete_by_chunk_id_and_node_name(chunk->fd.id, node_name);
+}
+
+void
+chunk_api_call_chunk_drop_replica_wrapper(ChunkMoveData *ccd, Oid serverid, bool transactional)
+{
+	if (transactional)
+		StartTransactionCommand();
+
+	chunk_api_call_chunk_drop_replica(ccd->chunk, ccd->src_node, serverid);
+
+	if (transactional)
+	{
+		NameData application_name;
+
+		snprintf(application_name.data,
```

Comment:
I have some doubts that this operation is safe to do (destination buffer overlaps with argument data)

Reply:
Agree that this doesn't look right.

Reply:
This is bogus, I meant to use ccd.operation_id.data as the input and to write into application_name.data. Will fix.